### DISCUSSION TITLE
Get LS version number within LS code? Main Autocue 2.2.5/2.3.0 differences?

### DISCUSSION THREAD

**Discussion by Moonbase59** [savonet/liquidsoap#4045] [Q&A]
Is there an easy way to verify the running Liquidsoap version number _within_ LS code, possibly using SemVer code I already have?

I’m thinking of making my Autocue integration auto-adapt to the version 2.2.5/2.3.0 differences, so it could work from the same code base, and we wouldn’t have to have two different versions of Autocue.

Anything else apart from the new cross start/end durations I’d have to be aware of?

**Comment by vitoyucepi** [comment]
Hi @Moonbase59,
I tried to use the `liquidsoap.version` to make my module compatible with 2.2.4 and 2.2.5, but it quickly became a typing nightmare.

**Reply by Moonbase59** [reply]
@vitoyucepi Thanks for this one! Looks like this works fine:

```ruby
%include "autocue.cue_file.liq"
if semver_compare(liquidsoap.version, "2.3.0") >= 0 then
  # do something
end
```

**Reply by smimram** [reply]
It's better to use `liquidsoap.version.at_least`, for instance
```
if liquidsoap.version.at_least("2.3.0") then
  ...
end
```
It implements correctly version comparison!
